File: DSA/DSA/Sorting/quick.py
List= [3,92,14,27,81,299,104,201,880,11, 234,21, 43,21,84, 62,48, 22, 59,62 ]
def Swap(List,right,left):
    firstElement= List[right]
    i,j= right, left
    while( j>i):
        while(firstElement>=List):
            if(firstElement<List[i]):
                break
            i+=1

        while(j>i ):
            if (firstElement>List[j]):
                break
            j-=1

        List[i],List[j]= List[j], List[i]
    
    
    List[i],List[right]= List[right], List[i]
    
    return i

def QuickSort(List, right,left):
    if left<= right: 
        return
                                   
    index= Swap(List, right,left)
    QuickSort(List,right,index-1)
    QuickSort(List, index+1,len(List)-1)

QuickSort(List, 0,len(List)-1)
for i in List:
    print(i,end=" ")

File: DSA/DSA/Sorting/quick.cpp
#include <bits/stdc++.h> 
using namespace std; 

int f(vector<int> &arr,int low,int high){
    int i=low, j= high;
    while(i<j ){
        while(arr[low]>=arr[i]  ){
            i++; 
        }
        while(arr[low]<=arr[j] && j>= low+1){
            j--; 
        }

        if(i< j) swap(arr[i],arr[j]);
        }
    swap(arr[low],arr[j]); 
    return j;
}

void quick_sort(vector<int> &arr, int low, int high){
    int index;
    if (low >= high) return;
    if (low< high){
        index= f(arr,low, high);
    }
    quick_sort(arr,low,index-1);
    quick_sort(arr,index+1,high);
}

int main(){
    vector<int> arr= {3,92,14,27,81,299,104,201,880,11, 234,21, 43,21,84, 62,48, 22, 59,62 }; 
    quick_sort(arr, 0, arr.size()-1); 
    for(auto it: arr){
        cout << it << " "; 
    }
    return 0; 
}